pkg/util/fileutil/writer.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Forked from kubernetes/pkg/volume/util/atomic_writer.go @fcee7a0105
+// to borrow the file/path validation. This does not write files atomically
+// as described in https://github.com/kubernetes/kubernetes/issues/18372 so
+// behavior may be different from watching or reloading
+// files when compared to K8S native Secrets.
+
+package fileutil
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"sigs.k8s.io/secrets-store-csi-driver/provider/v1alpha1"
+)
+
+const (
+	maxFileNameLength = 255
```

Comment:
For Azure Key vault, the max object name length is 128 and I assume this is a reasonable limit for the Google Secrets Manager as well. Is there a restriction for Hashicorp vault? Since this limit is being imposed now we should make sure we aren't breaking any exist cases.

Reply:
It does not seem like the vault provider includes an restrictions on path lengths: https://github.com/hashicorp/vault-csi-provider/blob/20173a0f069739c2e9236c3b23bbe98a4af2c3ba/internal/provider/provider.go#L254

The GCP plugin has a length of 253 since `IsConfigMapKey` is currently used:
https://github.com/kubernetes/apimachinery/blob/2456ebdaba229616fab2161a615148884b46644b/pkg/util/validation/validation.go#L424

Reply:
@tomhjp Do you see anything breaking with imposing this limit on the secret name length?

Reply:
There's no explicit limit currently but I'm ok with introducing this limit 👍